modules/ROOT/pages/rate-limiting.adoc

```diff
@@ -0,0 +1,219 @@
+
+Let’s look how the same configuration of 3 requests every 10 seconds works when the configuration is clusterized and identifiers are added.
+
+=== Configuring Identifiers with Regular String
+
+When you add identifiers to the policy configuration from the UI, you can define groups of requests. The configured limits apply independently for each group. You can also use the identifier `#[attributes.method]` for one bucket per HTTP method in your Rate Limiting policy configuration. 
+
+//If the universe of possible requests can be  split into a finite number of groups, then each group can be assigned with a selector key. that resolves to a selector key (a String). One Rate Limit algorithm will be created per key.  
+
+An `Identifier` is a non-obligatory parameter.  By default the Identifier is blank. Based on how you configure the Identifier, the policy performs in the following ways:
+
+* When not configured, the Rate Limiting quota applies to every request per bucket or group.
+* When configured for an obligatory HTTP header, each case-sensitive value of the header has its own quota. 
++
+Quotas are created using the _lazy creation_ strategy.
+* When configured  for a non-obligatory HTTP header, custom header, payload, query parameter or expression, each value has its own quota. 
++
+If the Identifier is not sent in the request, the Identifier defaults to an empty value, having its own quota. This behavior allows the Rate Limiting policy to be applied to an API consumed by uncontrolled clients, and at the same time accommodates special buckets for the clients sending the Identifier (which may be internal controlled services).
+
+The following example shows the order of events that occur over a period of time using the Identifier `#[attributes.method]` for a limit of 3 requests every 10 seconds:
+
+image::rate-limiting-configure-identifiers-example.png[Rate Limiting Configure Identifiers Example,70%,85%]
+In the example:
+
+* Every HTTP method is allowed 3 requests every 10 seconds (in this example only GET and POST requests have been made to the API).
+* The Rate Limiting policy works in a fixed window fashion (see the window size bracket).
+* The window start times are independent.
+* The engine uses a _lazy creation_ strategy that spools a rate limiting algorithm whenever the first request for a method is received.
+
+=== Configuring Identifiers with DataWeave Expressions
+
+The rate limiting engine, which is HTTP agnostic, depends solely on the resolution of the DataWeave expression. You can alter the identifier expression to cover complex rate limiting scenarios.
+
+For example, you can configure a Rate Limiting policy with an Identifier that uses one bucket for all class A & C LAN requests, and another bucket for everything else. The following image illustrates the latter case, which corresponds to 3 requests per 10 seconds quota with the DataWeave expression #[attributes.queryParam[‘customIdentifier’]] as the policy identifier:
```

Comment:
Does "...The following image illustrates the latter case" refer to the second bucket in the previous sentence?
Because the previous sentence sounds like ONE case in which you have TWO buckets, not two cases....

Reply:
Copyedits only:
```suggestion
For example, you can configure a Rate Limiting policy with an Identifier that uses one bucket for all class A & C LAN requests and another bucket for everything else. The following image illustrates the latter case, which corresponds to a 3 requests per 10 seconds quota with the DataWeave expression #[attributes.queryParam[‘customIdentifier’]] as the policy identifier:
```

Reply:
BREAK HERE

Reply:
Done! Thanks!